### ISSUE TITLE
hw07 ready for grading

### ISSUE THREAD

**Issue by suminwei2772** [suminwei2772/STAT545#7]
Latest commit: f030ef1

[Makefile](https://github.com/suminwei2772/STAT545-547-hw-Wei-Lisa/blob/master/hw07/Makefile) shows how each R script is executed.

[hw07_make.md](https://github.com/suminwei2772/STAT545-547-hw-Wei-Lisa/blob/master/hw07/hw07_make.md) is the final report.

**Comment by dorawyy** [comment]
Hi Lisa,

Glad to review your homework :) Overall, I think you did a good job.
* The `README` file for your entire repo is clear, organized, and informative, which directs me to the right place of hw07 folder;
* You did great work in each step of your analysis, having different ways of data presentations, like tables, histogram, barplot.

Here are some suggestions:
* You successfully managed complicated dependencies within the Makefile, however, it is a bit difficult for me to follow from the first place. It would be quite helpful if you include a dependency graph for your project, and you can use `makefile2graph` to implement. Here are some instructions in homework07.

> _Consider including an image showing a graphical view (the dependency diagram) of your pipeline using `makefile2graph`. On Mac or Linux you can install `makefile2graph` using `Homebrew` or `Linuxbrew` with the command brew install `makefile2graph`._

* There might be some minor mistakes in running Rscript. You only need to provide the `.R` file as `RScript` command parameter, for example, use:
```
Rscript step2_read_data_analysis.R
```
rather than
```
Rscript step2_read_data_analysis.R words.en.crlf.txt
```

Hope these could be helpful!

Cheers,
-Yingying

**Comment by emwest** [comment]
I agree with the first reviewer, your repository is easy to navigate and intuitively set up. Your code is clean and easy to follow and I appreciate you using a dataset other than gapminder, choosing your own  path. I would have like to see a few more examples of useful data manipulations, although that is not exactly what the assignment was. 

Overall, you met all the expectations of a solid homework submission. Nice job.

**Comment by ksedivyhaley** [comment]
**Three or more scripts, an Rmd, and a Makefile:** Yes
**Starts by downloading data, ends with Rmd:** Yes
**The output of each is the modified input of the previous step:** Yes
**Includes some analysis and at least one figure:** Yes
**Makefile includes all scripts and Rmd with correct dependencies:** No (see comments)
**Makefile runs:** Yes
**Bonus (novelty, externalization, embedded figs, import data, pandoc/latex):** Non-gapminder dataset
**Reflection:** No

**Comments:**

- Since step3 script needs to read both words_length.txt and words_starts.txt, n_starts.png is dependent on both text files – even though it only actually uses the data from words_starts.
- The Rmd uses and therefore is dependent on n_starts.png in addition to length_distribution.png
- You can actually specify two outputs on the same line to make the Makefile more concise, eg:
```
length_distribution.png n_starts.png: words_length.txt words_starts.txt step3_draw_fig.R
	Rscript step3_draw_fig.R words_length.txt words_starts.txt
```
- I see you are using `args <- commandArgs(TRUE)` and `read_tsv(args[1], col_names="Words")` and then passing file names as arguments in your Makefile. This is interesting and I can see some potential advantages but it also makes it harder to read the code – why did you decide to do this for this script?
- It looks like your Rmd re-calculates words_length.txt and words_starts.txt instead of reading in the files. Why did you decide to do that?

*Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.*
